airflow/executors/kubernetes_executor.py

```diff
@@ -820,8 +877,14 @@ def _flush_result_queue(self):
             except Empty:
                 break
 
-    def end(self):
+    def end(self) -> None:
         """Called when the executor shuts down"""
+        if not self.task_queue:
+            raise AirflowException(NOT_STARTED_MESSAGE)
+        if not self.result_queue:
+            raise AirflowException(NOT_STARTED_MESSAGE)
```

Comment:
All these feel like it should be a custom subclass instead

Reply:
I fixed it in another way. I just moved the queue initialisation to ... __init__ :). it was set to None there and only initialised during the start() method. But indeed it makes sense to set the type of those queues as not-optional and initialise them in __init__()